addons/point_of_sale/models/pos_session.py

```diff
@@ -325,6 +326,15 @@ def _validate_session(self, balancing_account=False, amount_to_balance=0, bank_p
             self.sudo()._post_statement_difference(cash_difference_before_statements)
             if self.move_id.line_ids:
                 self.move_id.sudo().with_company(self.company_id)._post()
+                #We need to write the price_subtotal and price_total here because if we do it earlier the compute functions will overwrite it here /account/models/account_move_line.py _compute_totals
+                for key, amount_data in data['sales'].items():
+                    account_id = key[0]
+                    tax_ids = [tax[0] for tax in key[2]]
+                    tax_tag_ids = list(key[3])
+                    self.move_id.line_ids.filtered(lambda line: line.account_id.id == account_id and line.tax_ids.ids == tax_ids and line.tax_tag_ids.ids == tax_tag_ids).write({
```

Comment:
This is error prone. It's likely to get the wrong mapping. perhaps aside from collecting the tax amounts, you should also get the reference of the created `account.move.line` that will be written with "price_total" and "price_subtotal".